src/stub.rs

```diff
@@ -92,14 +92,15 @@ pub enum Cmd {
         variables: Vec<VariableCommand>,
     },
     Write {
-        text: String,
```

Comment:
Feels a bit excessive to store a Vec when we know we are not going to change the `text` anymore.

Yet it does make sense for the renderer, so I guess it should be fine.

Also have you considered merging Write and WriteJoin, treating Write as a WriteJoin with a vector with only one JoinTerm::Literal? I don't know what will result, I'm just curious.